spring-security/src/main/java/com/epam/esm/orders/authComponent/AuthOrderComponent.java

```diff
@@ -0,0 +1,27 @@
+package com.epam.esm.orders.authComponent;
+
+import com.epam.esm.exceptionhandler.exception.ServerException;
+import com.epam.esm.role.Role;
+import com.epam.esm.user.model.User;
+import com.epam.esm.utils.validation.DataValidation;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class AuthOrderComponent {
+
+    public boolean hasPermission(User user, Long id) {
+        log.debug("Validation request model Order ID " + id);
+
+        if (DataValidation.moreThenZero(id)) {
+            if(user.getOrders() !=null){
+                return user.getOrdersIds().contains(id) || user.getRole().equals(Role.ADMIN);
+            }
+            return user.getRole().equals(Role.ADMIN);
```

Comment:
I think you can add method that checks role to User to increase readability
e.g. `user.is(Role.ADMIN)`

Reply:
remade